tracetypes/org.eclipse.tracecompass.incubator.uftrace.core/src/org/eclipse/tracecompass/incubator/internal/uftrace/core/analysis/UfCallstackProvider.java

```diff
@@ -108,4 +108,10 @@ protected long getThreadId(@NonNull ITmfEvent event) {
         Integer resolve = fTidAspect.resolve(event);
         return resolve == null ? -1 : resolve.longValue();
     }
+
+    @Override
```

Comment:
It should make use of the ThreadNameAspect. Currently UFTrace has an ExecAspect which is named "ThreadName" because it inherits from the ThreadNameAspect base class. Then if the ExecAspect is corrected and the ThreadNameAspect correctly implemented we should be able to have both showing correctly and this method should no longer be needed.

Reply:
Fixed